hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java

```diff
@@ -60,15 +60,15 @@ public static OMDirectoryResult verifyFilesInPath(
       String dbDirKeyName = omMetadataManager.getOzoneDirKey(volumeName,
           bucketName, pathName);
 
-      if (omMetadataManager.getKeyTable().get(dbKeyName) != null) {
+      if (omMetadataManager.getKeyTable().isExist(dbKeyName)) {
```

Comment:
This patch looks correct, but I have a question about the earlier patch that went in to support this.

The documentation of keyMayExist -- says this.

_public boolean keyMayExist(ColumnFamilyHandle columnFamilyHandle,
                  byte[] key,
                  java.lang.StringBuilder value)
If the key definitely does not exist in the database, then this method returns false, else true. This check is potentially lighter-weight than invoking DB::Get(). One way to make this lighter weight is to avoid doing any IOs.
Parameters:
columnFamilyHandle - ColumnFamilyHandle instance
key - byte array of a key to search for
value - StringBuilder instance which is a out parameter if a value is found in block-cache._

There is a value parameter in the call, which is documented as -- out parameter if the value is found in the block cache.

The objective of calling into this function, I suppose to leverage that -- Mind you the keyMayExist is not bloom filter lookup, it says If the value is in memory already you get access to it.

However, if you read the code -- it looks like we ignore the possiblity of gettting the value back from the block cache.

Here is the code that was committed via 
"HDDS-1691 : RDBTable#isExist should use Rocksdb#keyMayExist"
`
      // RocksDB#keyMayExist
      // If the key definitely does not exist in the database, then this
      // method returns false, else true.
      return db.keyMayExist(handle, key, new StringBuilder())
          && db.get(handle, key) != null;`

So how does this call eliminate the IO cost, since the new StringBuilder is completely ignored ?

Reply:
Thank you @anuengineer. This is a valid observation. I will fix the working of the isExist method in https://github.com/apache/hadoop-ozone/pull/101 where we are modifying the implementation.

Reply:
yes, thanks @anuengineer for flagging it. Yes in true case if we get value from block cache we can use that.

Reply:
perfect, let me go ahead and commit this patch, since you will fix that issue in the #101 pull request.